src/transport/SessionManager.cpp

```diff
@@ -461,6 +461,18 @@ bool SessionManager::MarkSessionsAsDefunct(const ScopedNodeId & node, const Opti
     return found;
 }
 
+void SessionManager::UpdateAllSessionsPeerAddress(const ScopedNodeId & node, const Transport::PeerAddress & addr)
+{
+    mSecureSessions.ForEachSession([&node, &addr](auto session) {
+        if (session->IsActiveSession() && session->GetPeer() == node &&
+            Transport::SecureSession::Type::kCASE == session->GetSecureSessionType())
+        {
+            session->AsSecureSession()->SetPeerAddress(addr);
```

Comment:
Why does this need `AsSecureSession`?  Pretty sure `session` is `SecureSession *` here.  I guess this got copied from `MarkSessionsAsDefunct`?

Reply:
Yeah copy and paste, but I don't know where I got that from originally.... :shrug: